src/Sulu/Component/SmartContent/Orm/DataProviderRepositoryTrait.php

```diff
@@ -58,29 +58,26 @@ private function findByFiltersIds($filters, $page, $pageSize, $limit, $locale, $
     {
         $parameter = [];
 
-        $queryBuilder = $this->createQueryBuilder('c')
-            ->select('c.id')
+        $queryBuilder = $this->createQueryBuilder('entity')
+            ->select('entity.id')
             ->distinct()
-            ->orderBy('c.id', 'ASC');
+            ->orderBy('entity.id', 'ASC');
 
-        $tagRelation = $this->appendTagsRelation($queryBuilder, 'c');
-        $categoryRelation = $this->appendCategoriesRelation($queryBuilder, 'c');
+        $tagRelation = $this->appendTagsRelation($queryBuilder, 'entity');
+        $categoryRelation = $this->appendCategoriesRelation($queryBuilder, 'entity');
 
-        if (\array_key_exists('sortBy', $filters) && \is_array($filters['sortBy'])) {
-            $sortMethod = \array_key_exists('sortMethod', $filters) ? $filters['sortMethod'] : 'asc';
-            $this->appendSortBy($filters['sortBy'], $sortMethod, $queryBuilder, 'c', $locale);
-
-            foreach ($filters['sortBy'] as $sortColumn) {
-                $queryBuilder->addSelect($sortColumn);
-            }
+        if (isset($filters['sortBy'])) {
+            $sortMethod = $filters['sortMethod'] ?? 'asc';
+            $this->appendSortBy($filters['sortBy'], $sortMethod, $queryBuilder, 'entity', $locale);
+            $queryBuilder->addSelect(str_contains($filters['sortBy'], '.') ? $filters['sortBy'] : 'entity' . '.' . $filters['sortBy']);
```

Comment:
i cant think about a reason why this is needed. going to remove it and try if the tests will still work 🙂

Reply:
the reason why this is needed is the `->distinct()` a few line above. we cannot remove that because returning duplicated ids would break pagination. therefore i added the select again